packages/ember-metal/lib/mixin.js

```diff
@@ -633,9 +633,10 @@ function _keys(ret, mixin, seen) {
   seen[guidFor(mixin)] = true;
 
   if (mixin.properties) {
-    var props = mixin.properties;
-    for (var key in props) {
-      if (props.hasOwnProperty(key)) { ret[key] = true; }
```

Comment:
the semantically the output values would be the same (although sort order can differ depending on many things...), but that isn't why i asked.

If props is always a fast object and never has inherited properties (which maybe we cant guarentee) then `for (let key in props) { ret[key] = true; }` would be by far the fastest option.